Anime-App/Feature/Anime/ViewModel/AnimeViewModel.swift

```diff
@@ -8,48 +8,44 @@
 import Foundation
 
 class AnimeViewModel {
-    func getAnime(completion: @escaping (Result<[Anime], Error>) -> Void) {
+    
+    var showSpinner = false
+    
+    private(set) var animes: [Anime] = []
+    
+    func getAnime(completion: @escaping (Result<Void,NetworkingError>) -> Void) {
         // get request
-        guard let url = URL(string: "https://kitsu.io/api/edge/anime") else { return }
-        URLSession.shared.dataTask(with: url) { data, response, error in
-            guard error == nil else {
-                completion(.failure(error!))
-                return
-            }
-            do {
-                guard let data = data else {return}
-                let decoder = JSONDecoder()
-                decoder.keyDecodingStrategy = .convertFromSnakeCase
-                let anime = try decoder.decode(AnimeResult.self, from: data)
-                completion(.success(anime.data))
-                print(anime)
-            } catch {
-                completion(.failure(error))
+        let url = UrlEndpoint.baseUrl.endpoint + UrlEndpoint.getAnime.endpoint
+        
+        NetworkingManager.shared.request(url, type: AnimeResult.self) { res in
+            self.showSpinner = true
```

Comment:
Also where is this even being used? I don't see it being used anywhere in the view controller, if you're trying to show and hide an spinner this isn't SwiftUI you can't observe changes and expect the UI to automatically refresh you have to tell it to do so, think about a way to communicate to the view controller that the spinner should be shown or hidden in the view controller